server/src/inmantals/server.py

```diff
@@ -118,22 +125,83 @@ def flatten(self, line, char):
         assert char < 100000
         return line * 100000 + char
 
+    def create_tmp_project(self) -> str:
+        ten_random_digits = "".join(random.choice(string.digits) for _ in range(10))
+        project_name = "project" + ten_random_digits
+        self.project_dir = os.path.join("/tmp", "vs-code", project_name)
+
+        os.makedirs(self.project_dir, exist_ok=True)
+        logger.info(f"project created at {self.project_dir}")
+
+        os.mkdir(os.path.join(self.project_dir, "libs"))
+
+        install_mode = InstallMode.master
+
+        modulepath = ["libs", os.path.dirname(self.rootPath)]
+
+        with open(os.path.join(self.project_dir, "project.yml"), "w+") as fd:
+            metadata: typing.Mapping[str, object] = {
+                "name": "testcase",
+                "description": "Project for testcase",
+                "repo": yaml.safe_load(self.repos),
```

Comment:
how do you know if it is a pip or git type repo? I think we need two config values

Reply:
The setting is a yaml list e.g.: `[{'url': 'https://github.com/inmanta/', 'type': 'git'}, {'url': 'https://pypi.org/simple', 'type': 'package'}]`

Reply:
I followed the doc https://docs.inmanta.com/inmanta-service-orchestrator/latest/reference/projectyml.html#project-yml

Reply:
oh, I would have never guessed , but I think that is good enough for now